internal/app/transport/requestv2.go

```diff
@@ -2,22 +2,25 @@ package transport
 
 import (
 	discoveryv2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	gcpv2 "github.com/envoyproxy/go-control-plane/pkg/cache/v2"
 	structpb "github.com/golang/protobuf/ptypes/struct"
 	status "google.golang.org/genproto/googleapis/rpc/status"
 )
 
 var _ Request = &RequestV2{}
 
 // NewRequestV2 creates a Request objects which wraps v2.DiscoveryRequest
-func NewRequestV2(r *discoveryv2.DiscoveryRequest) *RequestV2 {
+func NewRequestV2(r *discoveryv2.DiscoveryRequest, resp chan<- gcpv2.Response) *RequestV2 {
```

Comment:
Can you explain a bit about the choice to have the resp chan passed as a param rather than being initialized here?

Reply:
The channel initialization is now done by go-control-plane and the channel is passed through `CreateWatch`. Therefore it cannot be initialized here.